website/static/js/registrationUtils.js

```diff
@@ -198,6 +204,18 @@ var Question = function(data, id) {
     });
 
     /**
+     * @returns {Boolean} true if the value <input> does validate
+     **/
+    self.hasErrors = ko.computed(function() {
+        if( self.required ){
```

Comment:
The problem with this is that the `hasErrors()` function is called in the mako template to change the color of the nav pill, regardless of if the question is required. This will throw `self.value.isValid is not a function`. One solution would be to create different `li` based on `if: $data.required` but this feels a bit verbose as it only involves one attribute of a ko css binding. Is there a way to create the basic `li` then assign the needed css attribute after? Eg: 

```html
<li data-bind="css: {
      registration-editor-question-current: $root.currentQuestion().id === $data.id,
      list-group-item-danger: $root.showValidation() && $data.validationStatus()
      },
     click: $root.currentQuestion.bind($root, $data)"
     class="registration-editor-question list-group-item">
     <a data-bind="attr.href: '#' + id, text: nav"></a>
</li>
<span data-bind="if: $data.required">
    li.css.append(list-item-warning: $data.hasErrors()
</span>